### PR TITLE
Added gmmk v2 p65 keyboard.

### PR DESCRIPTION
**Pull request by itarze**
## Description
Assist Glorious to add a GMMK V2 65% keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/gmmk/gmmk2/p65/iso/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+#define _BL 0
+#define _FL 1
```
Comment (drashna) [inline keyboards/gmmk/gmmk2/p65/iso/keymaps/default/keymap.c:22]:
```suggestion
enum custom_layers {
    _BL,
    _FL
};
```

File: keyboards/gmmk/gmmk2/p65/iso/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+#define _BL 0
+#define _FL 1
```
Comment (drashna) [inline keyboards/gmmk/gmmk2/p65/iso/keymaps/via/keymap.c:22]:
```suggestion
enum custom_layers {
    _BL,
    _FL
};
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by guilleortas** [comment]
Hi @itarze, I have a couple of questions. Why changing the EEPROM driver from custom (see [rules.mk](https://github.com/GloriousThrall/qmk_firmware/blob/glorious_gmmk2/keyboards/gmmk/gmmk2/p65/iso/rules.mk)) to `wear_leveling`?
Also, do you know what's the maximum size of the bin? I'm planning to build VIAL for the GMMK 2 65%, which might require more space.

**Comment by itarze** [comment]
used/total: 44k/256k @guilleortas

**Comment by guilleortas** [comment]
@itarze great news, thanks. I'll try it out as soon as my GMMK2 arrives, should be tomorrow.
I was checking the [datasheet of the WB32F3G71XX](http://www.westberrytech.com/down/20210903/WB32F3G71-DS.pdf) and it says the flash size is 128kB rather than 256kB, any ideas?

**Comment by itarze** [comment]
WB32F3G71xC is 256k. @guilleortas

**Comment by guilleortas** [comment]
@itarze I have tested your branch and everything works beautifully on my GMMK 2 65% ISO, except that for some reason VIA doesn't pick it up, even with the appropriate design file (I made it myself). VIAL GUI has the same issue and shows as GMMK Pro, sideloading the VIA design file has no effect.
Am I missing something? Could you provide the correct json design file for VIA?
Thanks you!

**Comment by itarze** [comment]
[gmmk_v2_p65_iso.zip](https://github.com/qmk/qmk_firmware/files/9462483/gmmk_v2_p65_iso.zip)
@guilleortas  You try it.

**Comment by guilleortas** [comment]
Thanks for you swift reply @itarze. Unfortunately it still shows as GMMK Pro, even in the key tester->test matrix view the layout is that of the GMMK Pro. I'm afraid the origin of this issue must be somewhere in the firmware, as VIAL GUI needs no json design file and still picks it up as the Pro.

**Comment by guilleortas** [comment]
@itarze Perfect, it worked great! Thank you for the swift reply.
I'm looking at the differences between out design file and other than the underglow effects I'm still not seeing why mine didn't work...
Just one more question, are you familiar with VIAL? I'm able to compile and flash VIAL and everything works great in the GUI configurator and all the features are picked up just fine, however the RGB is not working. Any ideas?

**Comment by itarze** [comment]
Do you mean that the current branch compiles the firmware, RGB does not work? @guilleortas

**Comment by guilleortas** [comment]
@itarze No, this PR works great for me (RGB included) now that your design file solved the VIA issue.
What I meant is that using this as a base, I ported the GMMK 2 to VIAL, and VIAL works fine, except that RGB doesn't work (I have made sure it wasn't just off).

**Comment by luizkowalski** [comment]
hey @itarze 
first of all, SOLID WORK on this one. I just compiled this firmware (for via) on mac and it worked perfectly! I can't believe it, really, awesome work dude
second, I have a couple of questions:

1. the VIA config file you posted has a typo
image

2. any chance you could remember the name of the underglow effect that "explodes" when you press a key? this one was present on the default firmware but not on the VIA config file you've attached

**Comment by luizkowalski** [comment]
I also noticed something weird (on mac, at least)
When I lock the screen, the keyboard "shuts down" after a minute or so after inactivity (RGB goes off)
when I press any button, the lights turn back on but the keyboard doesn't respond to any key press, I have to reconnect the cable. any idea what could be the problem?

**Comment by itarze** [comment]
> I also noticed something weird (on mac, at least) When I lock the screen, the keyboard "shuts down" after a minute or so after inactivity (RGB goes off) when I press any button, the lights turn back on but the keyboard doesn't respond to any key press, I have to reconnect the cable. any idea what could be the problem?

@luizkowalski  Add this PR change to the firmware.

**Comment by itarze** [comment]
> hey @itarze first of all, SOLID WORK on this one. I just compiled this firmware (for via) on mac and it worked perfectly! I can't believe it, really, awesome work dude second, I have a couple of questions:
> 
> 1. the VIA config file you posted has a typo
> 
> image
> 
> 2. any chance you could remember the name of the underglow effect that "explodes" when you press a key? this one was present on the default firmware but not on the VIA config file you've attached

1. Thank you very much for reminding me, I have corrected the typo.
2. Could you please provide a video so that I can better find this effect?

**Comment by luizkowalski** [comment]
> 2. Could you please provide a video so that I can better find this effect?

yes., sure, it is this one: https://www.youtube.com/shorts/88_bw3COPhY

I managed to find it, simply by doing this:

* Updated `config.h` and added this

```c
#define RGB_MATRIX_KEYPRESSES
#define RGB_MATRIX_FRAMEBUFFER_EFFECTS
```

then started to navigate through the effects with `fn+right` until I found it

I also copied the changes from that PR you mentioned and flashed it again, the issue is gone 🎉

**Comment by itarze** [comment]
@luizkowalski 
Good, [gmmk_v2_p65.json](https://github.com/qmk/qmk_firmware/files/9477065/gmmk_v2_p65.zip).
I have added all the RGB effects to VIA JSON.
> ```c
> #define RGB_MATRIX_KEYPRESSES
> #define RGB_MATRIX_FRAMEBUFFER_EFFECTS
> ```
Define these two macros to enable full effects.

**Comment by luizkowalski** [comment]
Yeah, I added these definitions and flashed it yesterday, worked perfectly!
I think you could add these to the PR too

**Review by zvecr — APPROVED** [review]

**Pull request merged**